src/fparser/two/utils.py

```diff
@@ -380,9 +411,45 @@ def __new__(cls, string, parent_cls=None):
             errmsg = u"{0}: '{1}'".format(cls.__name__, string)
         raise NoMatchError(errmsg)
 
+    def get_root(self):
+        '''
+        Returns the root of the parse tree to which this node belongs.
+
+        :returns: the node at the root of the tree.
+        :rtype: :py:class:`fparser.two.utils.Base`
+
+        '''
+        parent = self
+        while parent.parent:
```

Comment:
I suggest storing the result the first time to avoid iterating on each call.

Reply:
As discussed, this could cause problems when manipulating a parse tree so I'm leaving it as dynamic. I'll keep it as `get_root()` to signal to the user that there's some cost associated with it.

Reply:
Agreed.